app/controllers/projects/blob_controller.rb

```diff
@@ -44,11 +50,16 @@ def update
     result = Files::UpdateService.new(@project, current_user, @commit_params).execute
 
     if result[:status] == :success
-      flash[:notice] = "Your changes have been successfully committed"
-      redirect_to after_edit_path
+      respond_to do |format|
+        format.html { redirect_to after_edit_path }
+        format.json { render json: { message: "success", filePath: after_edit_path } }
+      end
     else
       flash[:alert] = result[:message]
-      render :edit
+      respond_to do |format|
+        format.html { render :edit }
+        format.json { render json: { message: "failed", filePath: request.referrer } }
```

Comment:
Actually, they are the same:

http://apidock.com/rails/v2.3.2/Rack/Request/referrer